### ISSUE TITLE
Integration with V2 Projects

### ISSUE THREAD

**Issue by e0d** [github/issue-metrics#57]
Firstly, thanks for this action, it's extremely useful.

However, I've been very excited to move our PR and issue management to V2 projects because I think that project statuses will be, over time, a better mechanism for managing issue work flow than labels.  [Here's the board](https://github.com/orgs/openedx/projects/19/views/1) we use for managing incoming contributions to help understand our workflow.  We want to measure each steps to detect slow downs and optimize flow.

Do you anticipate that this action will support "time in status" for V2 projects?

[I don't think this works today](https://github.com/orgs/community/discussions/57326), so perhaps supporting this is currently blocked?

**Comment by zkoppert** [comment]
Wow, what a great use case! Didn't even think of measuring time in a project column before. Am I correct in my understanding that when you say "time in status" that is equivalent to "time in column x on project board y"?

I can check in on the API capabilities to see if we could add this functionality. I'll report back here when I know more!

**Comment by e0d** [comment]
Exactly.  Columns in V2 projects can relate to status field options -- I believe this is the default.

<img width="321" alt="image" src="https://github.com/github/issue-metrics/assets/1270274/daf1c086-1f96-49b9-9b55-e8cf721288b0">

Per project, you can represent the status options that are appropriate.

<img width="1077" alt="image" src="https://github.com/github/issue-metrics/assets/1270274/4e84b658-b93d-4bd1-b89d-d210427738b6">

Issues and PRs can be represented on multiple projects simultaneously with different status options per board.

**Comment by e0d** [comment]
@zkoppert can you provide any context on how this is blocked or whether that situation is likely to change?  We may have to undertake a fair bit of work to work around if it's going to be a while.

**Comment by zkoppert** [comment]
Like the communities link post in the description describes, the timeline events aren't working correctly for projects. I've made a request to the team but it is not being prioritized at this time. Unfortunately I don't have a estimate on when this may be addressed but I would expect not soon.

**Comment by e0d** [comment]
Understood.  Thanks for the reply it's helpful to know this isn't a priority.

**Comment by mlabeeb03** [comment]
I used this script to make `status_events` list which is similar to the `label_events` list we already have. Perhaps we could use this until the API supports it?
```
status_events = []
    html_content = requests.get(issue.issue.html_url)._content
    soup = BeautifulSoup(html_content, 'html.parser')
    script_tag = soup.find('script', {'type': 'application/json', 'data-target': 'react-app.embeddedData'})
    json_data = {}
    if script_tag:
        json_data = json.loads(script_tag.string)
    if json_data:
        for edge in json_data["payload"]["preloadedQueries"][0]["result"]["data"]["repository"]["issue"]["frontTimelineItems"]["edges"]:
            node = edge["node"]
            if "status" in node:
                if node["status"] in statuses:         
                    status_events.append({"createdAt": datetime.strptime((node["createdAt"]), "%Y-%m-%dT%H:%M:%SZ").replace(tzinfo=timezone.utc), "event": "statused", "statusName": node["status"]})
                if node["previousStatus"] != "" and node["previousStatus"] in statuses:
                    status_events.append({"createdAt": datetime.strptime((node["createdAt"]), "%Y-%m-%dT%H:%M:%SZ").replace(tzinfo=timezone.utc), "event": "unstatused", "statusName": node["previousStatus"]})
    return status_events
```
